tools/static_check_containers

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# use github internal registry if available, e.g. within github actions
+if [ "$(curl -s -w "%{http_code}" https://ghcr.io/v2/ -o /dev/null)" = "401" ]; then
+    img=hadolint/hadolint
+else
+    img=ghcr.io/hadolint/hadolint
```

Comment:
I don't know if/why the image thing is useful (it's not used by the GHA). But this script should probably be moved to the "common" repo then

Reply:
On GHA ghcr.io/hadolint/hadolint should be used to save bandwidth on GHA side and hence runtime for us

Reply:
Maybe a comment could be added to mention that. I was just surprised since "the" GHA for hadolint doesn't seem to be using it and other uses of it I came across didn't. If this makes things faster, awesome 👍🏾

Reply:
Well, the run in GHA would automatically fallback to "hadolint/hadolint" so we don't ensure the one on ghcr.io is actually used. Better to double-check